src/interfaces/chain.h

```diff
@@ -160,6 +163,9 @@ class Chain
         int64_t* time = nullptr,
         int64_t* max_time = nullptr) = 0;
 
+    //! Get unspent outputs associated with a transaction.
+    virtual std::vector<Coin> findCoins(const std::vector<COutPoint>& outputs) = 0;
```

Comment:
I wonder if this should return an `std::vector<std::pair<COutPoint, Coin>>`? Otherwise could note in the comment that result follows `outputs` order.